File: Estudos/Prova2/Bee1258/Main.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

// Estrutura para armazenar os dados de cada pedido
typedef struct {
    char nome[100];
    char cor[20];
    char tamanho;
} Camiseta;

// Função de comparação para o qsort
int compare(const void* a, const void* b) {
    Camiseta* c1 = (Camiseta*)a;
    Camiseta* c2 = (Camiseta*)b;
    
    // 1º Critério: Cor ascendente
    int comp_cor = strcmp(c1->cor, c2->cor);
    if (comp_cor != 0) return comp_cor;
    
    // 2º Critério: Tamanho decrescente ('P' > 'M' > 'G' na tabela ASCII)
    if (c1->tamanho != c2->tamanho) {
        return (c2->tamanho - c1->tamanho);
    }
    
    // 3º Critério: Nome ascendente
    return strcmp(c1->nome, c2->nome);
}

int main() {
    int n;
    int primeiro = 1;
    
    // O programa termina quando N for igual a 0
    while (scanf("%d", &n) == 1 && n != 0) {
        // Imprime linha em branco ENTRE os casos de teste
        if (!primeiro) {
            printf("\n");
        }
        primeiro = 0;
        
        // Aloca dinamicamente o vetor com base na quantidade N fornecida
        Camiseta* camisetas = (Camiseta*)malloc(n * sizeof(Camiseta));
        
        for (int i = 0; i < n; i++) {
            // " %[^\n]" lê toda a linha do nome ignorando quebras de linha anteriores
            scanf(" %[^\n]", camisetas[i].nome); 
            scanf("%s %c", camisetas[i].cor, &camisetas[i].tamanho);
        }
        
        // Ordena o vetor usando o algoritmo Quicksort nativo do C
        qsort(camisetas, n, sizeof(Camiseta), compare);
        
        // Impressão dos resultados conforme a especificação
        for (int i = 0; i < n; i++) {
            printf("%s %c %s\n", camisetas[i].cor, camisetas[i].tamanho, camisetas[i].nome);
        }
        
        free(camisetas); // Libera a memória alocada
    }
    
    return 0;
}

File: Estudos/Prova2/Bee1258/Main.java
import java.util.Arrays;
import java.util.Scanner;

public class Main {
    
    // Classe interna para representar o pedido de camiseta
    static class Camiseta {
        String nome;
        String cor;
        char tamanho;

        Camiseta(String nome, String cor, char tamanho) {
            this.nome = nome;
            this.cor = cor;
            this.tamanho = tamanho;
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        boolean primeiro = true;

        while (scanner.hasNextLine()) {
            String linhaN = scanner.nextLine().trim();
            if (linhaN.isEmpty()) continue;
            
            int n = Integer.parseInt(linhaN);
            if (n == 0) break; // Condição de parada

            // Imprime linha em branco ENTRE os casos de teste
            if (!primeiro) {
                System.out.println();
            }
            primeiro = false;

            // Uso de vetor puro de tamanho fixo N
            Camiseta[] camisetas = new Camiseta[n];

            for (int i = 0; i < n; i++) {
                String nome = scanner.nextLine().trim();
                
                // Lê a linha com a cor e o tamanho e divide pelo espaço em branco
                String[] detalhes = scanner.nextLine().trim().split("\\s+");
                String cor = detalhes[0];
                char tamanho = detalhes[1].charAt(0);

                camisetas[i] = new Camiseta(nome, cor, tamanho);
            }

            // Ordenação personalizada utilizando Arrays.sort
            Arrays.sort(camisetas, (c1, c2) -> {
                // 1º Critério: Cor ascendente
                int compCor = c1.cor.compareTo(c2.cor);
                if (compCor != 0) return compCor;
                
                // 2º Critério: Tamanho decrescente (c2 comparado com c1 inverte a ordem)
                if (c1.tamanho != c2.tamanho) {
                    return Character.compare(c2.tamanho, c1.tamanho);
                }
                
                // 3º Critério: Nome ascendente
                return c1.nome.compareTo(c2.nome);
            });

            // Impressão dos resultados
            for (Camiseta c : camisetas) {
                System.out.println(c.cor + " " + c.tamanho + " " + c.nome);
            }
        }
        
        scanner.close();
    }
}
